pkg/giturl/giturl.go

```diff
@@ -121,6 +121,8 @@ func GenerateLink(repo, commit, file string, line int64) string {
 	// Some paths contain '%' which breaks |url.Parse| if not encoded.
 	// https://developer.mozilla.org/en-US/docs/Glossary/Percent-encoding
 	file = strings.ReplaceAll(file, "%", "%25")
+	file = strings.ReplaceAll(file, "[", "%5B")
+	file = strings.ReplaceAll(file, "]", "%5D")
```

Comment:
Good catch on handling these special characters. For this kind of URL encoding, we can also use the standard library's [net/url.PathEscape()](https://pkg.go.dev/net/url#PathEscape) function instead of manually replacing special characters.